src/main/kotlin/com/mparticle/kits/IterableKit.kt

```diff
@@ -155,36 +162,43 @@ class IterableKit : KitIntegration(), ActivityListener, ApplicationStateListener
                     id = MParticle.getInstance()?.Identity()?.deviceApplicationStamp
                 }
             }
-            if (id != null) {
-                "$id@placeholder.email"
-            } else {
-                null
-            }
+            id
+        }
+    }
+
+    private fun String?.getPlaceholderEmail(): String? = this?.let { "$it@placeholder.email" }
+
+    private fun handleOnSuccess(userId: String?, mParticleUser: MParticleUser) {
+        if (prefersUserId) {
+            IterableApi.getInstance().setUserId(userId)
+        } else {
+            val userIdentities = mParticleUser.userIdentities
+            val mpEmail = userIdentities[IdentityType.Email]
+            val placeholderEmail = userId.getPlaceholderEmail()
```

Comment:
...does this compile?

Reply:
pretty cool if so :-). doesnt' have to be this PR - but we should decide our pattern/style here.

Reply:
Yes :) So basically is just getting the map from mParticuleUser, then a value from the map and userId.getPlaceholderEmail() is a private extension String function I wrote: 

 private fun String?.getPlaceholderEmail(): String? = this?.let { "$it@placeholder.email" }